Terminal.Gui/Configuration/ScopeJsonConverter.cs

```diff
@@ -1,140 +1,189 @@
-				var property = scope!.GetType ().GetProperties ().Where (p => {
-					var jia = p.GetCustomAttribute (typeof (JsonIncludeAttribute)) as JsonIncludeAttribute;
-					if (jia != null) {
-						var jpna = p.GetCustomAttribute (typeof (JsonPropertyNameAttribute)) as JsonPropertyNameAttribute;
-						if (jpna?.Name == propertyName) {
-							// Bit of a hack, modifying propertyName in an enumerator...
-							propertyName = p.Name;
-							return true;
-						}
-
-						return p.Name == propertyName;
-					}
-					return false;
-				}).FirstOrDefault ();
-
-				if (property != null) {
-					var prop = scope.GetType ().GetProperty (propertyName!)!;
-					prop.SetValue (scope, JsonSerializer.Deserialize (ref reader, prop.PropertyType, options));
-				} else {
-					// Unknown property
-					throw new JsonException ($"Unknown property name \"{propertyName}\".");
-				}
-			}
-		}
-		throw new JsonException ();
-	}
-
-	public override void Write (Utf8JsonWriter writer, scopeT scope, JsonSerializerOptions options)
-	{
-		writer.WriteStartObject ();
-
-		var properties = scope!.GetType ().GetProperties ().Where (p => p.GetCustomAttribute (typeof (JsonIncludeAttribute)) != null);
-		foreach (var p in properties) {
-			writer.WritePropertyName (ConfigProperty.GetJsonPropertyName (p));
-			JsonSerializer.Serialize (writer, scope.GetType ().GetProperty (p.Name)?.GetValue (scope), options);
-		}
-
-		foreach (var p in from p in scope
-					.Where (cp =>
-					cp.Value.PropertyInfo?.GetCustomAttribute (typeof (SerializableConfigurationProperty)) is
-					SerializableConfigurationProperty scp && scp?.Scope == typeof (scopeT))
-					where p.Value.PropertyValue != null
-					select p) {
-
-			writer.WritePropertyName (p.Key);
-			var propertyType = p.Value.PropertyInfo?.PropertyType;
-
-			if (propertyType != null && p.Value.PropertyInfo?.GetCustomAttribute (typeof (JsonConverterAttribute)) is JsonConverterAttribute jca) {
-				var converter = Activator.CreateInstance (jca.ConverterType!)!;
-				if (converter.GetType ().BaseType == typeof (JsonConverterFactory)) {
-					var factory = (JsonConverterFactory)converter;
-					if (factory.CanConvert (propertyType)) {
-						converter = factory.CreateConverter (propertyType, options)!;
-					}
-				}
-				if (p.Value.PropertyValue != null) {
-					converter.GetType ().GetMethod ("Write")?.Invoke (converter, new object [] { writer, p.Value.PropertyValue, options });
-				}
-			} else {
-				JsonSerializer.Serialize (writer, p.Value.PropertyValue, options);
-			}
-		}
-		writer.WriteEndObject ();
-	}
+    public override scopeT Read (ref Utf8JsonReader reader, Type typeToConvert, JsonSerializerOptions options) {
+        if (reader.TokenType != JsonTokenType.StartObject) {
+            throw new JsonException (
+                $"Expected a JSON object (\"{{ \"propName\" : ... }}\"), but got \"{reader.TokenType}\"."
+            );
+        }
+
+        var scope = (scopeT)Activator.CreateInstance (typeof (scopeT))!;
```

Comment:
(out of scope)

This should be generic all the way, as it calls the default constructor anyway. Therefore, the scopeT (which should also be TScope, by convention) should have a type parameter constraint including `new()` and then this line and similar should just call `new TScope()`. Important for AoT, too.